### PR TITLE
[SPARK-11735] [SQL] Add a check in the constructor of SQLContext to make sure its SparkContext is not stopped

### PR DESCRIPTION
**Pull request by yhuai**
https://issues.apache.org/jira/browse/SPARK-11735

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 1]
```diff
@@ -78,6 +78,9 @@ class SQLContext private[sql](
   }
   def this(sparkContext: JavaSparkContext) = this(sparkContext.sc)
 
+  // Make sure the sparkContext we have at here is not stopped.
+  sparkContext.assertNotStopped
```
Comment (yhuai) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: @pwendell @rxin @marmbrus 

@mengxr and I had a quick discussion. We think it is generally good to make sure the `sparkContext` provided to a `SQLContext` is not stopped. To do so, we'd like change the access modifier of `assertNotStopped` to `private[spark]`. Is this change good?

Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: that's fine - can you add parenthesis

**Comment by rxin** [comment]
Changes LGTM - can you bring it up to date and fix the style violation?

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by yhuai** [comment]
We probably need to fix python test.

**Comment by andrewor14** [comment]
retest this please

**Comment by gatorsmile** [comment]
retest this please

**Comment by 3ourroom** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

3ourroom@naver.com 님께 보내신 메일 <Re: [spark] [SPARK-11735] [SQL] Add a check in the constructor of SQLContext to make sure its SparkContext is not stopped (#9702)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
@yhuai looks like this has a legit failure in python

**Comment by yhuai** [comment]
it is. I just did not get a chance to fix it.

**Pull request closed without merging**